File: backend/app/schemas/data_source.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class DataSourceBase(BaseModel):
    name: str
    connector_type: str


class DataSourceCreate(DataSourceBase):
    pass


class DataSourceResponse(DataSourceBase):
    model_config = ConfigDict(from_attributes=True)

    id: str
    sync_status: str
    sync_health_pct: int
    tables_schema: str
    sync_logs: str
    last_synced_at: datetime | None = None

File: frontend/src/types/dataSource.ts
export type ConnectorType = 'zendesk' | 'salesforce' | 'shopify' | 'slack' | 'kb_repo';
export type SyncStatus = 'healthy' | 'degraded' | 'error';

export interface DataSourceResponse {
  id: string;
  name: string;
  connector_type: ConnectorType;
  sync_status: SyncStatus;
  sync_health_pct: number;
  tables_schema: string;
  sync_logs: string;
  last_synced_at: string;
}
